_source/_posts/2022-01-18-deploy-dotnet-container-aws-fargate.md

```diff
@@ -0,0 +1,640 @@
+
+This kind of management reliability continues into data management, memory management, and any other policies that need to be enforced. This ultimately creates a great deal of business agility as operations can reliably open the doors to developers knowing that they have some established rails. As for size, without the need for a whole OS, your container images are often only a few MB, vs. many GB. On the topic of speed, since there isn't a complete boot cycle, the container can effectively start up as fast as the hosted application can start.
+
+The difficulty with containers moves further up the abstraction chain (as is the style of such things) than with Virtual Machines. The challenge comes down to networking and managing container images across many resources so that you can treat a set of computing as one homogeneous unit. Fortunately, there are many solutions for this with products like Kubernetes, OpenShift, Mesos, Nomad, and others. But with this, you are ultimately back to having to manage a fleet of computers along with all the management overhead that goes along with it.
+
+This is the sweet spot of AWS Elastic Container Service (ECS) Fargate. Fargate gives you networking abstractions across a virtual network known as a VPC (Virtual Private Cloud). This network abstraction is built right into the heart of AWS and is well vetted for any type of workload, including high-security government workloads. Fargate takes this a step further by abstracting away the machine management. You can set up traditional clusters and manage your machines if you want, but leveraging Fargate simplifies one more part of your process. Ultimately our goal with using cloud vendors in the first place is to let them be good at infrastructure management so we can be good at managing our business.
+
+Time to jump into it and try it out using .NET containers with AWS Fargate!
+
+**What you will need to get started**
+
+What you'll need to continue are the following:
+
+- Basic knowledge of .NET
+- [.NET SDK 3.1](https://docs.microsoft.com/en-in/dotnet/core/install/)
+- [An AWS account (we'll be using a free tier product)](https://aws.amazon.com/free)
+- [Okta CLI](https://cli.okta.com)
+- [AWS CLI V2](https://aws.amazon.com/cli/)
+- [Docker Desktop for Windows/macOS (Not required if you are on Linux)](https://www.docker.com/get-started)
+
+This tutorial assumes you already have Docker set up and running.
+
+## Build and Dockerize a .NET Chat Application
+
+So what do you want to build? To keep things simple so we can see the value in using something other than standard HTTP protocols, I'll use SignalR to build a very basic chat application.
+
+- Secure: Only logged-in clients should be able to use the chat functionality
+- Chat users names must come from their validated identity
+- Real-time
+
+To achieve this, I am going to use four technologies:
```

Comment:
The upcoming section goes forward using first-person "I" - our InfoDev Style Guide recommends very limited use of "I" and "we" in favor of you. I don't advise any rewriting, so this guidance for going forward. Have a look here: https://oktawiki.atlassian.net/wiki/spaces/DOC/pages/722929640/InfoDev+Style+Guide (under Voice, and their pointer to "Person" here: https://docs.microsoft.com/en-us/style-guide/grammar/person

Reply:
corrected